pkg/docker/util.go

```diff
@@ -54,27 +64,88 @@ func GetImageRegistryAuth(auths *client.AuthConfigurations, imageName string) cl
 		glog.V(5).Infof("Using %s credentials for pulling %s", auth.Email, imageName)
 		return auth
 	}
-	return client.AuthConfiguration{}
+	return api.AuthConfig{}
 }
 
 // LoadImageRegistryAuth loads and returns the set of client auth objects from a docker config
 // json file.
-func LoadImageRegistryAuth(dockerCfg io.Reader) *client.AuthConfigurations {
-	auths, err := client.NewAuthConfigurations(dockerCfg)
+func LoadImageRegistryAuth(dockerCfg io.Reader) *AuthConfigurations {
+	auths, err := NewAuthConfigurations(dockerCfg)
 	if err != nil {
 		glog.V(0).Infof("error: Unable to load docker config")
 		return nil
 	}
 	return auths
 }
 
+// begin borrowed from fsouza
```

Comment:
And the code they use to hydrate it isn't public?  The reason i'm worried is that docker has previously changed the format of the config.json file and broken us, so i'd rather depend on an api for reading this file than doing it ourselves.